### ISSUE TITLE
The setup

### ISSUE THREAD

**Issue by Shikamaru5** [lucidrains/imagen-pytorch#295]
This is what I have so far the first one is the training script.

      from imagen_pytorch import Unet, Imagen, ImagenTrainer
      from imagen_pytorch.data import Dataset
      import torch
      
      # unets for unconditional imagen
      
      unet = Unet(
          dim = 32,
          dim_mults = (1, 2, 4, 8),
          num_resnet_blocks = 1,
          layer_attns = (False, False, False, True),
          layer_cross_attns = False
      )
      
      # imagen, which contains the unet above
      
      imagen = Imagen(
          unets = unet,
          text_encoder_name = 't5-large',
          image_sizes = 64,
          timesteps = 5_000
      )
      
      trainer = ImagenTrainer(
          imagen,
          fp16 = True,
          split_valid_from_train = True,
          split_valid_fraction = 0.05,
      ).cuda()
      
      text_embeds = torch.randn(64, 256, 1024).cuda()
      images = torch.randn(64, 3, 256, 256).cuda()
      
      # instantiate your dataloader, which returns the necessary inputs to the DDPM as tuple in the order of images, text embeddings, then text masks. in this case, only images is returned as it is unconditional training
      trainer.load('./bigger.pt')
      
      train_ds = Dataset('cities', image_size = 64)
      
      trainer.add_train_dataset(train_ds, batch_size = 32)
      
      # working training loop
      
      
      loss = trainer(images, text_embeds = text_embeds, unet_number = 1, max_batch_size = 8)
      print(f'loss: {loss}')
      
      trainer.save('./bigger.pt')

I initially was working with the unconditional script but I figure it might be better to be conditioned on text. I've upped the time_steps to 5000, and when I run my script I call train.py --epoches 1000. Idk if that's a good epoch number but that's what I've done.
The second is the generator script.

      import torch
      from imagen_pytorch import Unet, Imagen, ImagenTrainer
      from torchvision.utils import save_image
      
      # unet for imagen
      
      unet = Unet(
          dim = 32,
          dim_mults = (1, 2, 4, 8),
          num_resnet_blocks = 1,
          layer_attns = (False, False, False, True),
          layer_cross_attns = False
      )
      
      # imagen, which contains the unet above
      
      imagen = Imagen(
          unets = unet,
          text_encoder_name = 't5-large',
          image_sizes = 64,
          timesteps = 5_000
      )
      
      trainer = ImagenTrainer(
          imagen,
          fp16 = True,
          split_valid_from_train = True,
          split_valid_fraction = 0.05,
      ).cuda()
      
      text_embeds = torch.randn(64, 256, 1024).cuda()
      
      trainer.load('./bigger.pt')
      
      images = imagen.sample(texts = [
        'a mural.'
      ], cond_scale = 5.)
      
      images.shape # (2, 3, 256, 256)
      
      for idx, image in enumerate(images):
        save_image(image, f"sample_{idx}.png")

I am using two datasets to start with, just so I can figure out what is best and even if it's grainy at least generate some sort of image. However, all I've gotten thus far is this rainbow static image, can someone please help me figure out what I'm doing wrong? I really can't figure out what I'm supposed to do from the read me docs to the comments it seems it's really all over the place when it comes to how this is supposed to be done. @lucidrains

**Comment by Shikamaru5** [comment]
So I've worked on it some more and tried to train the first and second unet on my datasets. I then changed the generator a bit to include the ElucidatedImagen but I've got the problem where Idk how I'm supposed to be loading the Unets because I had to train them seperately. Thus they have different checkpoints, and when I start up the script it states that it failed loading state dict, attempting partial load. It then loads but I still get that pretty staticky image in place of an actual image. Any assistance would be greatly appreciated.
![sample_0](https://user-images.githubusercontent.com/86502093/209480490-65b7466b-00b8-4281-adad-4192fceb06ed.png)

**Comment by Heite92** [comment]
It seems like after you added the dataset to the trainer you still train with the random noise images, thus you get random noise images.

`loss = trainer(images, text_embeds = text_embeds, unet_number = 1, max_batch_size = 8)`

instead you want to call (take a look at the dataloader ssample)

`for i in range(200000):
    loss = trainer.train_step(unet_number = 1, max_batch_size = 4)`
